feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/ConversationViewModel.kt

```diff
@@ -715,4 +724,71 @@ internal open class ConversationViewModel(
             imageProvider.createImageLoader(context, authorizationToken)
         }
     }
+
+    fun onFileSelected(uri: Uri?, fileName: String, context: Context) {
+        Log.d("ConversationViewModel", "File selected: $uri")
+        viewModelScope.launch {
+            _selectedFileUri.emit(uri)
+            _selectedFileName.emit(fileName)
+            uploadFileOrImage(context)
+        }
+    }
+
+    private fun uploadFileOrImage(context: Context) {
+        viewModelScope.launch(coroutineContext) {
+            try {
+                val fileBytes = withContext(Dispatchers.IO) {
+                    _selectedFileUri.value?.let { uri ->
+                        context.contentResolver.openInputStream(uri)?.use { inputStream ->
+                            val fileSize = inputStream.available()
+
+                            val maxFileSize = 5 * 1024 * 1024
+                            if (fileSize > maxFileSize) {
+                                throw IllegalArgumentException("The file size exceeds the limit of 5MB.")
+                            }
+                            inputStream.readBytes()
+                        }
+                    } ?: throw IllegalArgumentException("No file selected")
+                }
+
+                val response = metisModificationService.uploadFileOrImage(
+                    context = metisContext,
+                    courseId = courseId,
+                    conversationId = conversationId,
+                    fileBytes = fileBytes,
+                    fileName = _selectedFileName.value,
+                    serverUrl = serverConfigurationService.serverUrl.first(),
+                    authToken = accountService.authToken.first()
+                )
+
+                val fileUploadResponse = (response as NetworkResponse.Response).data
+                val filePath = fileUploadResponse.path
+                    ?: throw IllegalArgumentException("There has been a prob")
+
+                val currentText = newMessageText.value.text
+                val updatedText = if (isImage(_selectedFileName.value)) {
+                    "$currentText\n![${_selectedFileName.value}]($filePath)\n" // Image markdown
+                } else {
+                    "$currentText\n[${_selectedFileName.value}]($filePath)\n"  // File markdown
+                }
+                newMessageText.value = TextFieldValue(updatedText)
+                clearSelectedFile()
+            } catch (e: Exception) {
+                clearSelectedFile()
+                withContext(Dispatchers.Main) {
+                    Toast.makeText(
```

Comment:
I think returning the error message to the user as toast is not a good user experience. You could return something like "An error occurred while uploading your file", for the other exceptions above I would introduce string resource values.

Reply:
What do you suggest instead of the toast, pop up or any other option you have in your mind.